src/emoji/__tests__/data-test.js

```diff
@@ -1,64 +1,89 @@
   });
-
   test('works for some multi-codepoint emoji', () => {
-    check('0',        '0⃣', '0\u{20e3}');
-    check('asterisk', '*⃣', '*\u{20e3}');
-    check('hash',     '#⃣', '#\u{20e3}');
+    check('0030-20e3', '0⃣', '0\u{20e3}');
+    check('002a-20e3', '*⃣', '*\u{20e3}');
+    check('0023-20e3', '#⃣', '#\u{20e3}');
   });
 });
 
-describe('getFilteredEmojiNames', () => {
+
+describe('getFilteredEmojis', () => {
   test('empty query returns all emojis', () => {
-    const list = getFilteredEmojiNames('', {});
+    const list = getFilteredEmojis('', {});
     expect(list).toHaveLength(1560);
   });
 
   test('non existing query returns empty list', () => {
-    const list = getFilteredEmojiNames('qwerty', {});
+    const list = getFilteredEmojis('qwerty', {});
     expect(list).toHaveLength(0);
   });
 
   test('returns a sorted list of emojis starting with query', () => {
-    const list = getFilteredEmojiNames('go', {});
+    const list = getFilteredEmojis('go', {});
     expect(list).toEqual([
-      'go',
-      'goal',
-      'goat',
-      'goblin',
-      'gold',
-      'gold_record',
-      'golf',
-      'gondola',
-      'goodnight',
-      'gooooooooal',
-      'gorilla',
-      'got_it',
+      { code: '1f3c1', name: 'go', type: 'unicode' },
+      { code: '1f945', name: 'goal', type: 'unicode' },
+      { code: '1f410', name: 'goat', type: 'unicode' },
+      { code: '1f47a', name: 'goblin', type: 'unicode' },
+      { code: '1f947', name: 'gold', type: 'unicode' },
+      { code: '1f4bd', name: 'gold_record', type: 'unicode' },
+      { code: '1f3cc', name: 'golf', type: 'unicode' },
+      { code: '1f6a0', name: 'gondola', type: 'unicode' },
+      { code: '1f31b', name: 'goodnight', type: 'unicode' },
+      { code: '1f945', name: 'gooooooooal', type: 'unicode' },
+      { code: '1f98d', name: 'gorilla', type: 'unicode' },
+      { code: '1f44c', name: 'got_it', type: 'unicode' },
     ]);
   });
 
-  test('search in realm emojis as well', () => {
-    expect(getFilteredEmojiNames('qwerty', { qwerty: {} })).toEqual(['qwerty']);
+  test('search in realm emoji as well', () => {
+    expect(
+      getFilteredEmojis('qwerty', {
+        qwerty: {
+          code: 'qwerty',
+          deactivated: false,
+          name: 'qwerty',
+          source_url: 'url'
+        }
+      })
+    ).toEqual([{ name: 'qwerty', code: 'qwerty', type: 'image' }]);
   });
 
   test('remove duplicates', () => {
-    expect(getFilteredEmojiNames('dog', {})).toEqual(['dog', 'dogi']);
-    expect(getFilteredEmojiNames('dog', { dog: {} })).toEqual(['dog', 'dogi']);
+    expect(getFilteredEmojis('dog', {})).toEqual([
+      { code: '1f415', name: 'dog', type: 'unicode' },
+      { code: '1f94b', name: 'dogi', type: 'unicode' }
+    ]);
+    expect(
+      getFilteredEmojis('dog', {
+        dog: {
+          code: 'dog',
+          deactivated: false,
+          name: 'qwerty',
+          source_url: 'url'
+        }
```

Comment:
Hmm, this looks odd.
* The name doesn't match the key. Those should always be equal in this data structure, right?
* The code is `dog`. But that's not the right format for the `code` in a realm-emoji reaction. Following the link in our jsdoc on `Reaction`, look at the comment on `emoji_code` [here](https://github.com/zulip/zulip/blob/master/zerver/models.py):

  > For realm emoji (aka user uploaded custom emoji), the ID (in ASCII decimal) of the RealmEmoji object.

  So this `code` should be a number like 123.

Ditto with `code: 'qwerty'` above.